src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,170 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null) {
+            head = node;
+            tail = head;
+            size++;
+        } else {
+            node = new Node<>(tail, value, null);
+            node.prev.next = node;
+            tail = node;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexRangeInAdd(index);
+        if (size == index) {
+            add(value);
+        } else {
+            Node<T> node = findNodeByIndex(index);
+            Node<T> newNode = new Node<>(node.prev, value, node);
+            if (node.prev == null) {
+                head = newNode;
+            } else {
+                node.prev.next = newNode;
+            }
+            node.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
```

Comment:
Consider using the enhanced for loop for better readability when iterating over the `list`.